### PR TITLE
AirfRANS: attention heads sweep — 4H vs 16H on champion config

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current AirfRANS champion uses 2L/256d/4H (64d per head). We have not yet tested whether a finer-grained attention with 16 heads (16d/head) can capture more localised flow features — particularly in the volume field where `vol_mse=0.002777` remains the harder target. Conversely, 4H (64d/head) gives each head more representational capacity per slot, which may help the model resolve smooth pressure gradients. This sweep holds all other champion hyperparameters fixed (AdamW, lr=6e-4, T_max=50, gc=1.0, WD=1e-2, EMA=0.999, Fourier features) and varies only head count to isolate its effect.

Target: drive `vol_mse` toward ~0.0017 while holding `surface_mse` ≤ 0.000459.

## Instructions

Run both trials in sequence. Use `--wandb_group af-attention-heads-sweep` so both runs are grouped in W&B.

**Trial 1 — 4 heads (64d/head, replicates current champion head count as a controlled baseline):**
```bash
cd target/icml2026 && python train.py --dataset airfrans --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 --ema-decay 0.999 --wandb_group af-attention-heads-sweep
```

**Trial 2 — 16 heads (16d/head, finer-grained attention):**
```bash
cd target/icml2026 && python train.py --dataset airfrans --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 16 --epochs 999 --ema-decay 0.999 --wandb_group af-attention-heads-sweep
```

**Mandatory locked config (do NOT change):**
- Optimizer: AdamW, lr=6e-4, WD=1e-2, grad-clip=1.0
- Cosine annealing: T_max=50
- EMA decay: 0.999 (use `--ema-decay 0.999`)
- Architecture: 2 layers, 256d hidden dim
- Fourier features: `--enable-fourier` required

**Report for each trial:**
- W&B run ID and link
- `val_primary/surface_mse` (best epoch)
- `full_val/volume_mse` (best epoch)
- Best epoch number

We care especially about `vol_mse` — if 16H improves it below 0.002777 even slightly, that is a win worth noting.

## Baseline

Current AirfRANS champion — PR #3050 (stark/af-ema-champion), W&B run `z6pry4b9`:

| Metric | Value |
|--------|-------|
| `val_primary/surface_mse` | **0.000459** (epoch 771) |
| `full_val/volume_mse` | **0.002777** |

Champion config: 2L/256d/4H, AdamW lr=6e-4, T_max=50, gc=1.0, WD=1e-2, EMA=0.999, Fourier features enabled.

Reproduce command:
```bash
cd target/icml2026 && python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 --ema-decay 0.999
```

**Known dead ends (do not revisit):** 3L depth (catastrophic), EMA < 0.999, T_max ≠ 50, vol-weight=10x.

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3169 because ADVISOR: Closing per issue #3283 directive: 'Stop broad depth/width/heads' for AirfRANS. Reassigning to priority lane..
